test/common.py

```diff
@@ -1061,17 +1061,25 @@ def check_dylink(self):
     if self.get_setting('MEMORY64') == 2:
       self.skipTest('dynamic linking not supported with MEMORY64=2')
 
-  def require_v8(self):
+  def get_v8(self):
+    """Return v8 engine, if one is configured, otherwise None"""
     if not config.V8_ENGINE or config.V8_ENGINE not in config.JS_ENGINES:
+      return None
+    return config.V8_ENGINE
+
+  def require_v8(self):
+    v8 = self.get_v8()
+    if not v8:
       if 'EMTEST_SKIP_V8' in os.environ:
         self.skipTest('test requires v8 and EMTEST_SKIP_V8 is set')
       else:
         self.fail('d8 required to run this test.  Use EMTEST_SKIP_V8 to skip')
-    self.require_engine(config.V8_ENGINE)
+    self.require_engine(v8)
     self.cflags.append('-sENVIRONMENT=shell')
 
   def get_nodejs(self):
-    if config.NODE_JS_TEST not in self.js_engines:
+    """Return nodejs engine, if one is configured, otherwise None"""
+    if config.NODE_JS_TEST not in config.JS_ENGINES:
```

Comment:
Why did `self` change to `config` here?

Reply:
These function get v8 or node if its available in the config.  The `self.js_engines` might be subset, or some specific engine that was configured, but these function i always want to return the engine if its available, regardless of the current `self.js_engines` which could be a subset.

Reply:
In practice I don't think it should make much differences because `self.js_engines` always starts of as copy of `config.JS_ENGINES` and is then limited on per-test based by things like `@require_node`

Reply:
I think you would only notice this change if you had a test that did `require_node` followed by `require_v8` or something complicated like that.